src/bind.py

```diff
@@ -196,37 +239,89 @@ def handle_relation_data(
         Returns:
             A resulting DNSRecordProviderData to put in the relation databag
         """
-        # Create staging area
-        tempdir = tempfile.mkdtemp()
-
-        # Write zone files
-        zones = {}
+        # Create zones list
+        zones: typing.List[Zone] = []
         for record_requirer_data, _ in relation_data:
-            zones.update(self._to_bind_zones(record_requirer_data))
-        logger.debug("ZONES: %s", zones)
-        for name, content in zones.items():
-            self._write(pathlib.Path(tempdir, f"db.{name}"), content)
-
-        # Write the named.conf file
-        self._write(
-            pathlib.Path(tempdir, "named.conf.local"), self._generate_named_conf_local(list(zones))
-        )
+            zones.extend(self._record_requirer_data_to_zones(record_requirer_data))
+
+        # Check for conflicts
+        _, conflicting = self._get_conflicts(zones)
+        if len(conflicting) > 0:
```

Comment:
I didnt get it: you create the DNS record if there is conflict, is that it?

Reply:
I see that it can be not easily understandable but no: I'm returning the dns_record_provider_data that will be containing the conflicts if there's a conflict. This data will go into the databag to inform the other charms about the issue